libs/designsystem/modal/src/modal-wrapper/config/modal-config.ts

```diff
@@ -1,11 +1,16 @@
 import { ActivatedRoute, Data, Route } from '@angular/router';
 import { Observable } from 'rxjs';
 
+import { AlertConfig } from '../../public_api';
 import { DrawerSupplementaryAction } from './drawer-supplementary-action';
 
 export type ModalFlavor = 'modal' | 'drawer' | 'compact';
 export type ModalSize = 'small' | 'medium' | 'large' | 'full-height';
 
+export type ShowAlertCallback =
+  | (() => Promise<boolean | AlertConfig>)
+  | (() => boolean | AlertConfig);
```

Comment:
Consider simplifying:
```suggestion
  | (() => boolean | AlertConfig | Promise<boolean | AlertConfig>);
```